### ISSUE TITLE
[BC Break][3.4.x] cannotBeEmpty is not applicable to concrete nodes

### ISSUE THREAD

**Issue by iltar** [symfony/symfony#24614]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | no
| BC Break report? | yes
| RFC?             | no
| Symfony version  | 3.4.x

This seems to be caused by d40e7e4c2de1c4468714de8006f84aee287aace8.

```                                                                                             
[RuntimeException]                                                                                
In ArrayNodeDefinition.php line 501:                                                                                                                                                                  
  ->requiresAtLeastOneElement() is not applicable to concrete nodes 
  at path "forward.handlers"                                                         
```

My guess is that this should be a deprecation rather than an exception for 3.4.

**Comment by stof** [comment]
Can you give the stack trace of the element, to know in which bundle this happens ?

**Comment by iltar** [comment]
This happened in 2 custom bundles, so it's easy to fix. As far as I can tell, the exceptions are 100% legit, just shouldn't break compatibility with with 3.3

**Comment by ro0NL** [comment]
@iltar you are calling `cannotBeEmpty()` then? Which had no effect before, but is now taken into account. 

To be clear; the behavior is correct right?  Throwing the exception can be seen as a new feature i guess, it improves your code no?  Anyway this is why i targeted 3.4 and definitely not lower branches :-)

ref https://github.com/symfony/symfony/pull/20361

**Comment by iltar** [comment]
@ro0NL yes, calling that function while it does nothing. I don't know if it's a new feature, it certainly breaks backwards compatibility as it didn't crash before.

**Comment by ro0NL** [comment]
The issue i want to solve is exactly this; ppl calling `cannotBeEmpty` while it has zero effect; without you really knowing it / being aware.

Given that i prefer the exception over deprecation; forcing the fix in 3.4. But maybe you're right about a deprecation path. Yet it may be ignored; thus leaves you with dead/wrong code.

**Comment by ro0NL** [comment]
Ill have a look ASAP to change the exception in favor of a deprecation.
